packages/components/TabList/src/Tab/Tab.tsx

```diff
@@ -74,7 +74,14 @@ export const Tab = compressible<TabProps, TabTokens>((props: TabProps, useTokens
       }
     });
 
-    const { icon, tabKey, ...mergedProps } = mergeProps(tab.props, final, {
+    // `onLayout` is unused and excluded from the rest of the mergedProps to be passed into the RootSlot.
+    // This is to ensure that the chained layout callback created in useTabAnimation isn't overwritten.
+    const {
+      icon,
+      tabKey,
+      onLayout: _,
```

Comment:
I think this is fine for now, and it's good that this comment is here, but we might get a request in the future to be able to handle children having onLayout working.

Reply:
This is a consequence of an `onLayout` callback passed from `React.cloneElement` completely overriding the hook's `onLayout` callback. I still don't fully understand why that occurs with cloning vs just passing an `onLayout` prop, but this at least is a working work-around.